### PR TITLE
cleanup build scripts and build only one set of images

### PR DESCRIPTION
**Pull request by rkpagadala**
Currently we builds 2 sets of images one of which points to gcr.io/istio-testing and another to docker.io/istio
This has been a great source of confusion and complexity.

With this change for
Daily Builds docker images are pushed to gcr.io/istio-release, istioctl and the yaml/helm files also point to  gcr.io/istio-release. This set of artifacts are then tested and published to 
https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/

Monthly Builds docker images are pushed to docker.io/istio, istioctl and the yaml/helm files also point to  docker.io/istio. This set of artifacts are then tested and published to github as a release

### CODE REVIEW

File: prow/istio-postsubmit.sh [hunk 1]
```diff
@@ -48,4 +48,4 @@ time ISTIO_DOCKER_HUB="${HUB}" make push HUB="${HUB}" TAG="${TAG}"
 
 cd "$ROOT"
 OUTDIR="$(mktemp -d /tmp/outdir.XXXX)"
-time (./release/cloud_builder.sh -p istio-release-pipeline-data/daily-build/postsubmit/"${GIT_SHA}" -q istio-release -t "${GIT_SHA}"  -o "${OUTDIR}" || echo "not failing make for now")
+time (./release/cloud_builder.sh -h "gcr.io/istio-testing" -t "${GIT_SHA}"  -o "${OUTDIR}" || echo "not failing make istio-archive for now")
```
Comment (hklai) [inline prow/istio-postsubmit.sh]: why is the trailing echo needed to prevent failure?

File: release/airflow/cloud_build.template.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
       ],
       "name": "gcr.io/istio-testing/istio-builder:0.5.10",
       "dir": "go/src/istio.io/istio",
-      "args": [ "./release/store_artifacts.sh", "-p", "$_GCS_PATH", "-q", "$_GCR_PATH", "-v", "$_VER_STRING", "-o", "/output" ]
+      "args": [ "./release/store_artifacts.sh", "-c", "$_BRANCH", "-p", "$_GCS_PATH", "-h", "$_DOCKER_HUB", "-v", "$_VER_STRING", "-o", "/output" ]
```
Comment (hklai) [inline release/airflow/cloud_build.template.json]: why is GCR_PATH removed in most of these files?

I suppose we still publish to GCR (along with pushing to dovkrthub), Just that istioctl hub is defaulted to dockerhub, right?

File: release/cloud_builder.sh [hunk 3]
```diff
@@ -81,30 +82,15 @@ MAKE_TARGETS=(istio-archive)
...
-if [[ -n "${TEST_DOCKER_HUB}" ]]; then
-# this copy is being done here instead of above because we
-# are conservative, if new artifacts are created we don't
-# inadvertently clobber them
-  cp "${OUTPUT_PATH}"/gcr.io/istio-*z* "${OUTPUT_PATH}/"
-fi
 
 if [ "${BUILD_DOCKER}" == "true" ]; then
+  # we always save the docker tars and point them to docker.io/istio
+  # later scripts retag the tars as needed and push to the docker/gcr as needed
```
[the remark this answers is not in the document]
Comment (rkpagadala) [reply release/cloud_builder.sh]: changed the wording a but

File: prow/istio-postsubmit.sh (hunk 1 above)
Comment (rkpagadala) [reply prow/istio-postsubmit.sh]: this will be moved to a different job, currently zip is not installed due to move to pod-utils

**Comment by rkpagadala** [comment]
@hklai PTAL updated description and comments in code

File: release/airflow/cloud_build.template.json (hunk 2 above)
Comment (hklai) [reply release/airflow/cloud_build.template.json]: If DOCKER_HUB can be GCR, should it be named as something like REGISTRY_PATH?

File: prow/istio-postsubmit.sh (hunk 1 above)
Comment (hklai) [reply prow/istio-postsubmit.sh]: Not sure I understand. cloud_builder.sh is currently failing due to not having zip installed? If so, why?

File: release/airflow/modify_values.sh [hunk 4]
```diff
@@ -27,6 +27,8 @@ function fix_values_yaml_worker() {
   local gcs_folder_path
   gcs_folder_path="$5"
 
+  echo "fixing  ${gcs_folder_path}/${tarball_name} with hub: ${HUB} tag: ${TAG}"
```
[the remark this answers is not in the document]
Comment (hklai) [reply release/airflow/modify_values.sh]: Please make it more clear in this comment.

**Comment by hklai** [comment]
> Monthly Builds docker images are pushed to docker.io/istio, istioctl and the yaml/helm files also point to docker.io/istio. This set of artifacts are then tested and published to github as a release

For monthly/LTS release, are we pushing to docker hub with the expected tag (e.g 1.1.0) to run tests?

**Comment by rkpagadala** [comment]
Yes we do. For monthly
1) push to docker.io/istio/pilot:1.1.0
2) test
3) release to github and tag

**Comment by costinm** [comment]
Do you have a build/test log using gcr.io and one with docker.io ? I'm curious how much faster gcr.io is 
compared with docker.io - I assume that's why we're doing the tests and daily builds on gcr, but is it a visible difference ? Because it would be simpler to just push to docker.io...

**Comment by hklai** [comment]
@costinm yup I have the same question too. I doubt speed is really a factor here, for something we do once or at most a few times a day.

Do you have problem pushing daily releases to docker.io/istio, or do you suggest to push to docker.io/istio-release for separation?

**Comment by rkpagadala** [comment]
@costinm the time difference is not that much its 3-4 min for gcr and 4-5 min for docker
the reason we push to different hubs is that docker.io/istio is our customer visible hub, we use different hub for pre-release docker images and for testing

**Pull request closed without merging**